src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -142,43 +125,32 @@ public async Task<bool> CanNavigateToPositionAsync(Workspace workspace, Document
                 workspace, documentId, vsTextSpan, cancellationToken).ConfigureAwait(false);
         }
 
-        public Task<INavigableLocation?> GetLocationForSpanAsync(
+        public async Task<INavigableLocation?> GetLocationForSpanAsync(
             Workspace workspace, DocumentId documentId, TextSpan textSpan, NavigationOptions options, bool allowInvalidSpan, CancellationToken cancellationToken)
         {
-            return GetNavigableLocationAsync(workspace,
+            if (!await this.CanNavigateToSpanAsync(workspace, documentId, textSpan, allowInvalidSpan, cancellationToken).ConfigureAwait(false))
+                return null;
```

Comment:
GetNavigableLocationAsync won't already return null if it's not navigable here?

Reply:
(comment applies to the other methods here)

Reply:
it should... but i wanted to make absolutely sure there was no disagreement on behavior so this helps enforce that.